view/frontend/templates/checkout/onepage/success/paynow_additional_info.phtml

```diff
@@ -1,5 +1,6 @@
-
-<p><?= __('PayNow QR Code.'); ?></p>
-<p style="margin:.5rem 0"><?= $block->escapeHtml(__('Amount to pay')) ?>: <strong><?= $block->getOrderAmount() ?></b></strong>
-<div><img src="data:image/png;base64, <?= $block->getPaynowQrcode() ?>" /></div>
+<?php if($block->getPaymentType() === 'paynow') : ?>
```

Comment:
It's a little confusing to me that the content of this file is specific to PayNow but we have still have to check that our payment type is PayNow; same with the PromptPay one.

Reply:
These all templates get included to layout page for all payment method types so checking is needed.